src/pages/EditRoomPage.tsx

```diff
@@ -0,0 +1,183 @@
+import useAlert from '../hooks/useAlert';
+import { useRoom } from '../hooks/useRoom';
+import { useNavigate, useParams, Link as RouterLink } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { DeleteForeverRounded, ViewListRounded } from '@mui/icons-material';
+import Room from '../room';
+import SaveButton from '../components/SaveButton/SaveButton';
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+
+function AutoCompleteDropdown(props: {readonly children?: React.ReactNode}) {
+    const theme = useTheme();
+    return (
+        <Paper sx={{ border: "1px solid " + theme.palette.primary.main, marginTop: 1 }}>
+            {props.children}
+        </Paper>
+    )
+}
+
+function EditRoomContent(props: { 
+    readonly room: Room, 
+    readonly labels: string[] }) {
+
+    const room = props.room;
+    const labels = props.labels;
+    const [mutatedRoom, setMutatedRoom] = useState<Room>(room);
+    const [isLoading, setIsloading] = useState(false);
+    const queryClient = useQueryClient();
+    const navigate = useNavigate();
+    const updateRoomMutation = useMutation({
+        mutationFn: (room: Room) => {
+            setIsloading(true);
+            return axios.put(`${import.meta.env.VITE_SRAMS_API_ADDRESS}room/updateRoom`, room);
+        },
+        onSuccess: () => {
+            setIsloading(false);
+        },
+    });
+    const mutate = () => {
+        updateRoomMutation.mutate(mutatedRoom);
+    }
+
+    const deleteRoomMutation = useMutation((id: string) => axios.delete(
+        `${import.meta.env.VITE_SRAMS_API_ADDRESS}room/deleteRoom`,
+        { params: { roomId: id } }
+      ), {
+        onSuccess: () => {
+            console.log("Room deleted");
+            queryClient.invalidateQueries("allRooms");
+            navigate("/admin")
+        }
+      });
+    const handleDelete = () => {
+        deleteRoomMutation.mutate(room.id);
+    }
+
+    //set initatl room state
+    useEffect(() => {
+        setMutatedRoom(room);
+    }, [room])
+
+    return (
+        <Card sx={{ padding: 4, overflow: "visible" }}>
+            <Stack direction={"column"} gap={4}>
+                <Stack direction={"row"} alignItems={'flex-start'} gap={2}>
+                    <TextField
+                        sx={{ flexGrow: 1 }}
+                        onChange={(e) => {
+                            const r = mutatedRoom;
+                            r.name = e.target.value;
+                            setMutatedRoom(r);
+                        }}
+                        label="Name"
+                        defaultValue={room.name}
+                        variant="standard" />
+                    <IconButton onClick={handleDelete}>
+                        <DeleteForeverRounded color='error' />
+                    </IconButton>
+                    <SaveButton saveHandler={mutate} isLoading={isLoading} />
```

Comment:
```suggestion
                    <IconButton onClick={handleDelete}>
                        <DeleteForeverRounded color='error' />
                    </IconButton>
                    <SaveButton saveHandler={mutate} isLoading={isLoading} />
```
Skal dette være en dropdown i stedet? Ellers kan man nemt klikke save og delete.
Ellers skal vi have en godkendelses knap når man klikker på dem.
Men ellers ser det meget godt ud. 🦧

Reply:
God pointe, jeg opretter et issue 👌